src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ArrayConverter.cs

```diff
@@ -9,9 +9,7 @@ namespace System.Text.Json.Serialization.Converters
     /// <summary>
     /// Converter for <cref>System.Array</cref>.
     /// </summary>
-    internal sealed class ArrayConverter<TCollection, TElement>
-        : IEnumerableDefaultConverter<TCollection, TElement>
-        where TCollection: IEnumerable
+    internal sealed class ArrayConverter<TCollection, TElement> : IEnumerableDefaultConverter<TElement[], TElement>
```

Comment:
It's not needed, however the converter factory infrastructure is making [a few assumptions](https://github.com/dotnet/runtime/blob/2053d653bfd7fc0a5d073a5d31e01600d2d5fa3f/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactory.cs#L179-L193) on how generic parameters are being used. I thought I'd preserve convention rather than special casing array converter construction.